code/game/gamemodes/gang/gang.dm

```diff
@@ -72,24 +139,108 @@ GLOBAL_LIST_INIT(gang_outfit_pool, list(/obj/item/clothing/suit/jacket/leather,/
 			boss.special_role = "[G.name] Gang [title]"
 			boss.restricted_roles = restricted_jobs
 			log_game("[boss.key] has been selected as the [title] for the [G.name] Gang")
+			gangpocalypse()
+	return 1
 
-	if(gangs.len < 2) //Need at least two gangs
-		return 0
 
-	return 1
+
+/datum/game_mode/gang/proc/gangpocalypse()
+	set waitfor = FALSE
+	if(target_captain)
+		for(var/turf/T in area_contents(target_captain))
```

Comment:
you should ```CHECK_TICK``` these outer loops (the ones going over turfs, the ones going over items should probably be fine)